horizon/facts/client.py

```diff
@@ -101,6 +101,26 @@ def convert_response(
                 headers=response.headers,
             )
 
+    @staticmethod
+    def extract_body(
+        response: HttpxResponse, expected_status_code: int = status.HTTP_200_OK
+    ):
+        if response.status_code != expected_status_code:
+            logger.warning(
+                f"Response status code is not {expected_status_code}, skipping wait for update."
+            )
+            return None
```

Comment:
I would skip the status code check because it requires you to actually know the status code and a change in the backend might break this.
Would do .raise_for_status() and catch if there is exception

Reply:
This actually breaks thing as some requests returns 201